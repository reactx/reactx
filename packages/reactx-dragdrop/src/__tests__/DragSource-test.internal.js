/**
 * Copyright (c) ReactX and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 *
 */

'use strict';
import ReactDOM from 'react-dom';

describe('while running in a browser environment', () => {
  let container;

  beforeEach(() => {
    jest.resetModules();

    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    ReactDOM.render(null, container);
    document.body.removeChild(container);
    container = null;
  });

  it('should render correctly', () => {
    // wrapper = (
    //   <div>
    //     <DragSource>
    //       <div>Drop Here</div>
    //     </DragSource>
    //   </div>
    // );
    // expect(ReactDOM.render(wrapper, container)).toMatchSnapshot();
  });

  it('should render correctly with provider', () => {
    // const dRef = React.createRef();
    // wrapper = (
    //   <DragDropProvider>
    //     <DragSource>
    //       <span>Drag Me!</span>
    //     </DragSource>
    //   </DragDropProvider>
    // );
    // ReactDOM.render(wrapper, container);
    // dRef.current.dispatchEvent(
    //   document.createEvent('MouseEvents').initEvent('mousedown', true, true),
    // );
    // wrapper = renderer
    //   .create(
    //     <DragDropProvider>
    //       <DragSource ref={dRef}>
    //         <span>Drag Me!</span>
    //       </DragSource>
    //     </DragDropProvider>,
    //   )
    //   .toJSON();
    // expect(wrapper).toMatchSnapshot();
  });

  it('should multiple and nested render correctly', () => {
    // wrapper = (
    //   <div>
    //     <DragSource>
    //       <div>
    //         <span>Drag Me!</span>
    //         <DragSource>
    //           <span>Drag Me!</span>
    //         </DragSource>
    //       </div>
    //     </DragSource>
    //     <DragSource>
    //       <span>Drag Me!</span>
    //     </DragSource>
    //   </div>
    // );
    // expect(ReactDOM.render(wrapper, container)).toMatchSnapshot();
  });

  describe('while running in a browser environment', () => {
    // it('should support onDragStart', () => {
    //   let divRef = React.createRef();
    //   let handleOnDragStart = jest.fn();
    //   function Component() {
    //     return (
    //       <DragDropProvider>
    //         <DragSource ref={divRef} onDragStart={handleOnDragStart}>
    //           <div>Drag me!</div>
    //         </DragSource>
    //       </DragDropProvider>
    //     );
    //   }
    //   ReactDOM.render(<Component />, container);
    //   const mouseOverEvent = document.createEvent('Event');
    //   mouseOverEvent.initEvent('dragstart', true, true);
    //   divRef.current.dispatchEvent(mouseOverEvent);
    //   expect(handleOnDragStart).toHaveBeenCalledTimes(1);
    // });
  });
});
