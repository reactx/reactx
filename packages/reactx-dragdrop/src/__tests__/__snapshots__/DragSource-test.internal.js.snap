// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while running in a browser environment should multiple and nested render correctly 1`] = `
<div>
  <div>
    <div>
      <span>
        Drag Me!
      </span>
      <div>
        <span>
          Drag Me!
        </span>
      </div>
    </div>
  </div>
  <div>
    <span>
      Drag Me!
    </span>
  </div>
</div>
`;

exports[`while running in a browser environment should render correctly 1`] = `
<div>
  <div>
    <div>
      Drop Here
    </div>
  </div>
</div>
`;
